OMEdit/OMEditGUI/Git/CommitChangesDialog.cpp

```diff
@@ -0,0 +1,254 @@
+  else if (status.trimmed().compare("D")== 0)
+    return "Deleted";
+  else if (status.trimmed().compare("R")== 0)
+    return "Renamed";
+  else if (status.trimmed().compare("C")== 0)
+    return "Copied";
+  else if (status.trimmed().compare("??")== 0)
+    return "Untracked";
+  else
+    // should never be reached
+    return "UnknownFileStatus";
+}
+
+/*!
+  Slot activated when mpCommitDescriptionTextBox textChanged signal is raised.\n
+  */
+void CommitChangesDialog::commitDescriptionTextChanged()
+{
+  if(!mpCommitDescriptionTextBox->toPlainText().isEmpty())
+     mpCommitButton->setEnabled(true);
+  else
+     mpCommitButton->setEnabled(false);
+}
+
+void CommitChangesDialog::commitFiles()
+{
+  QString filePath = MainWindow::instance()->getModelWidgetContainer()->getCurrentModelWidget()->getLibraryTreeItem()->getFileName();
+  QString nameStructure = MainWindow::instance()->getModelWidgetContainer()->getCurrentModelWidget()->getLibraryTreeItem()->getNameStructure();
+  QFileInfo info(filePath);
+  MainWindow::instance()->getGitCommands()->commitFiles(filePath, mpCommitDescriptionTextBox->toPlainText());
+
+  foreach (const QString &fileName, mpModifiedFiles) {
+     if(!fileName.isEmpty()) {
+       QString activity = getFileStatus(fileName.mid(0, 2));
+       QString fileURI = fileName.mid(3);
+       generateTraceabilityURI(nameStructure, info.absolutePath(), activity, fileURI );
+       }
+    }
+  accept();
+}
+
+void CommitChangesDialog::generateTraceabilityURI(QString nameStructure, QString filePath, QString activity, QString fileURI)
+{
+
+  QString toolURI, fileNameURI, activityURI, agentURI;
+  QFile file(filePath + "/" + nameStructure +".md");
+
+//  QFileInfo fi("C:/Users/alame60/Desktop/INTOCPS/traceability/OpenModelica/Models/BouncingBall.mo");
+//  QString base = fi.baseName();
+//  qDebug()<<"base name"<<base;
+//  QFile file(base +".md");
+
+  /*
+   * If file not exist it will create
+   * */
+  if (!file.open(QIODevice::WriteOnly | QIODevice::Text ))
+  {
+      //qDebug() << "FAIL TO CREATE FILE / FILE NOT EXIST***";
+  }
+
+  /*for Reading line by line from text file
+  while (!file.atEnd()) {
+      QByteArray line = file.readLine();
+      qDebug() << "read output - " << line;
+  }*/
+    QDateTime time = QDateTime::currentDateTime();
+
+  /*for writing line by line to text file */
+      fileNameURI = "Entity.file:" + fileURI + "#"+ MainWindow::instance()->getGitCommands()->getGitHash(MainWindow::instance()->getModelWidgetContainer()->getCurrentModelWidget()->getLibraryTreeItem()->getFileName());
+      toolURI = "Entity.softwareTool:" + MainWindow::instance()->getOMCProxy()->getVersion();
+      agentURI = "Agent:" + mpAuthorTextBox->text();
+      activityURI = "Activity." + activity +":" + time.toString("yyyy-MM-dd-hh-mm-ss");
+      QTextStream stream(&file);
+      stream << fileNameURI << "," << activityURI << "," << agentURI <<"," << toolURI;
+      file.flush();
+      file.close();
+}
```

Comment:
The above code looks  a bit strange. You should check if file is created successfully. Otherwise calling `flush` and `close` could lead to a crash.